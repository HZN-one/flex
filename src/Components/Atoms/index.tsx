export * from "./FAButton";
export * from "./Icon";
export * from "../Molecules/FMFormTextField";
export * from "./FATextField";
export * from "../Molecules/FMSelect";
export * from "./FACheckbox";
export * from "./FASwitch";
export * from "./FARadio";
export * from "./FAPagination";
export * from "./FAChip";
export * from "./FATooltip";
export * from "./FAFormControlLabel";
export * from "./FAFormControl";
