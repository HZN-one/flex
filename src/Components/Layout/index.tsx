export * from "./Initiator";
export * from "./FLSettingsNavbar";
