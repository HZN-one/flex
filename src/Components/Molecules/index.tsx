export * from "./FMAccordion";
export * from "./FMSearch";
export * from "./FMCard";
export * from "./FMStepper";
export * from "./FMTableSkeleton";
export * from "./FMTimeline";
export * from "./FMLabelValue";
export * from "./FMTabsMenu";
export * from "./FMInputAvatar";
