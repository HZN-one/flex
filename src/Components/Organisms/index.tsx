export * from "./FOCardWithImage";
export * from "./FOCardWithStepper";
export * from "./FOCardWithTextField";
export * from "./FODialogWithAlert";
export * from "./FOHeader";
export * from "./FOPaperRow";
export * from "./FOSideBar";
