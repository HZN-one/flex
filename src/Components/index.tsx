export * from './Atoms'
export * from './Layout'
