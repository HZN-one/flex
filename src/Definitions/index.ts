export * from "./Theme";
export * from "./constants";
export * from "./helpers";
