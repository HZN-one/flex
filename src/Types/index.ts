export type ElementPosition = "start" | "center" | "end";
