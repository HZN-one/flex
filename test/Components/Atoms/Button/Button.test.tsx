/**
 * @jest-environment jsdom
 */

import * as React from 'react'
import '@testing-library/jest-dom'

import { ButtonPegasus } from '../../../../src/Components'

import { fireEvent, render } from '@testing-library/react'

describe('ButtonPegasus', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ButtonPegasus testID="button-example">Test</ButtonPegasus>
    )
    expect(container).toMatchSnapshot()
  })

  it('should render with children prop', () => {
    const { getByText } = render(
      <ButtonPegasus testID="button-example">hey</ButtonPegasus>
    )

    expect(getByText('hey')).toBeTruthy()
  })

  it('should increment number on click', () => {
    let number = 1
    const { getByTestId } = render(
      <ButtonPegasus
        testID="button-example"
        onClick={() => {
          number += 1
        }}
      >
        ButtonPegasus
      </ButtonPegasus>
    )

    fireEvent.click(getByTestId('button-example'), new MouseEvent('click'))

    expect(number).toBe(2)
  })

  it('should render gradient blue button by gradientColor property', () => {
    const { getByTestId } = render(
      <ButtonPegasus testID="button-example" gradientColor="blue">
        ButtonPegasus
      </ButtonPegasus>
    )

    expect(getByTestId('button-example')).toHaveStyle(
      'background: linear-gradient(45deg, #2196F3 30%, #21CBF3 90%)'
    )
  })

  it('should render gradient red button by gradientColor property', () => {
    const { getByTestId } = render(
      <ButtonPegasus testID="button-example" gradientColor="red">
        ButtonPegasus
      </ButtonPegasus>
    )

    expect(getByTestId('button-example')).toHaveStyle(
      'background: linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)'
    )
  })
})
