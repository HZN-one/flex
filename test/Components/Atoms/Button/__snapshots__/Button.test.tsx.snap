// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPegasus should match snapshot 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    data-testid="button-example"
    tabindex="0"
    type="button"
  >
    Test
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
